### DISCUSSION TITLE
Two website questions

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#13833] [Archive]
Hi all!

We're planning on using Directus as our backend for our website. Two questions:

1. We want our contact form to post to a directus collection. In the Public role we set access on the ability to create an item in directus. My question is: what is the best practice of making sure this endpoint isn't ddos-ed and pulls down the rest of our backend? We have the rate limiter setup by the way.
2. Is there a way we can have the images (directus_files) come from CDN? We store them in google cloud storage.
Cheers!

Kees.

**Comment by rijkvanzanten** [comment]
> We want our contact form to post to a directus collection. In the Public role we set access on the ability to create an item in directus. My question is: what is the best practice of making sure this endpoint isn't ddos-ed and pulls down the rest of our backend? We have the rate limiter setup by the way.

The rate limiter is a start, but if you require more security, you might want to consider introducing some form of captcha. This does require you to either have a middleman server to verify the captcha and then post to Directus on the user's behalf, a custom endpoint, or a Flow that checks against this.

> Is there a way we can have the images (directus_files) come from CDN? We store them in google cloud storage.
Cheers!

I'd generally recommend putting the CDN in front of Directus itself. CDNs generally offer functionality to protect against DDoS attacks as well, and it means that you're able to still rely on the transformations in the /assets endpoint. Alternatively, you could open up the access to whatever file storage you're using, and hook up a CDN directly to that, but at that point you're effectively circumventing the usage of Directus completely, so you'd lose file access control and transformations 👍🏻

**Reply by keesvanbemmel** [reply]
Thanks @rijkvanzanten as always!
